File: sbe/binance/spot_3_2/generated-python/spot_sbe/PriceFilter.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple
from io import BytesIO

class PriceFilter:
    """SBE message: PriceFilter."""

    TEMPLATE_ID = 1
    SCHEMA_ID = 3
    SCHEMA_VERSION = 2
    BLOCK_LENGTH = 25

    def __init__(self):
        self.filter_type = None
        self.price_exponent = None
        self.min_price = None
        self.max_price = None
        self.tick_size = None

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()

        if self.price_exponent is not None:
            buffer.write(struct.pack('<b', self.price_exponent))
        if self.min_price is not None:
            buffer.write(struct.pack('<q', self.min_price))
        if self.max_price is not None:
            buffer.write(struct.pack('<q', self.max_price))
        if self.tick_size is not None:
            buffer.write(struct.pack('<q', self.tick_size))

        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        buffer = BytesIO(data)

        self.price_exponent = struct.unpack('<b', buffer.read(1))[0]
        self.min_price = struct.unpack('<q', buffer.read(8))[0]
        self.max_price = struct.unpack('<q', buffer.read(8))[0]
        self.tick_size = struct.unpack('<q', buffer.read(8))[0]

File: sbe/binance/spot_3_2/generated-php/spot_sbe/PriceFilter.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class PriceFilter
{
    public const TEMPLATE_ID = 1;
    public const SCHEMA_ID = 3;
    public const SCHEMA_VERSION = 2;
    public const BLOCK_LENGTH = 25;

    public int|float|array|null $filterType = null;
    public int|float|array|null $priceExponent = null;
    public int|float|array|null $minPrice = null;
    public int|float|array|null $maxPrice = null;
    public int|float|array|null $tickSize = null;

    public function encode(): string
    {
        $buffer = '';

        if ($this->priceExponent !== null) {
            $buffer .= pack('c', $this->priceExponent);
        }
        if ($this->minPrice !== null) {
            $buffer .= pack('q', $this->minPrice);
        }
        if ($this->maxPrice !== null) {
            $buffer .= pack('q', $this->maxPrice);
        }
        if ($this->tickSize !== null) {
            $buffer .= pack('q', $this->tickSize);
        }

        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;

        $this->priceExponent = unpack('c', substr($data, $offset, 1))[1];
        $offset += 1;
        $this->minPrice = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->maxPrice = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->tickSize = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
    }
}
